theano/sandbox/cuda/opt.py

```diff
@@ -2399,6 +2399,34 @@ def local_gpu_extract_diagonal(node):
     return False
 
 
+@register_opt()
+@local_optimizer([theano.tensor.Diagonal])
```

Comment:
I thought ExtractDiag allows offset to be non-zero, while Diagonal dosen't, so I kept it. After looking the code again I find it worng... Both of them don't support non-default offsets now. So I will merge the two opts.